cli/cmd/cmds/devx.go

```diff
@@ -0,0 +1,207 @@
+package cmds
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"unicode"
+
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/text"
+
+	"github.com/input-output-hk/catalyst-forge/cli/pkg/run"
+)
+
+type DevX struct {
+	MarkdownPath string `arg:"" help:"Path to the markdown file."`
+	CommandName  string `arg:"" help:"Command to be executed."`
+}
+
+func (c *DevX) Run(ctx run.RunContext, logger *slog.Logger) error {
+	// read the file from the specified path
+	raw, err := os.ReadFile(c.MarkdownPath)
+	if err != nil {
+		return fmt.Errorf("could not read file at %s: %v", c.MarkdownPath, err)
+	}
+
+	// parse the file with prepared options
+	commandGroups, err := extractCommandGroups(raw)
+	if err != nil {
+		return err
+	}
+
+	// exec the command
+	return processCmd(commandGroups, c.CommandName)
+}
+
+type command struct {
+	content  string
+	lang     *string
+	platform *string
+}
+
+func (cmd *command) Exec() error {
+	executorCmd, executorArgs := getLangExecutor(cmd.lang)
+	if executorCmd == "" {
+		return fmt.Errorf("only commands running with `sh` can be executed")
+	}
+
+	// check if the command is available
+	if _, err := exec.LookPath(executorCmd); err != nil {
+		return fmt.Errorf("command '%s' not found in PATH", executorCmd)
+	}
+
+	// start executing the command
+	execCmd := exec.Command(executorCmd, formatArgs(executorArgs, cmd.content)...)
```

Comment:
Take a peek at the [code here](https://github.com/input-output-hk/catalyst-forge/blob/master/cli/pkg/executor/local.go). You should be able to instantiate one of these with the `WithRedirect()` option and it will accomplish the same thing as redirecting the output back to the user. See the example instantiation [here](https://github.com/input-output-hk/catalyst-forge/blob/7de57df949094a71acfe4ec6940b1d7b44ccd72a/cli/cmd/cmds/run.go#L32).